sdk/servicebus/azure-servicebus/azure/servicebus/common/message.py

```diff
@@ -429,64 +293,121 @@ def __init__(self, message):
         super(PeekMessage, self).__init__(None, message=message)
 
     @property
-    def locked_until(self):
-        raise TypeError("Peeked message is not locked.")
+    def settled(self):
+        """Whether the message has been settled.
 
-    @property
-    def lock_token(self):
-        raise TypeError("Peeked message is not locked.")
+        This will aways be `True` for a message received using ReceiveAndDelete mode,
+        otherwise it will be `False` until the message is completed or otherwise settled.
 
-    def renew_lock(self):
-        """A PeekMessage cannot be renewed. Raises `TypeError`."""
-        raise TypeError("Peeked message is not locked.")
+        :rtype: bool
+        """
+        return self.message.settled
 
-    def complete(self):
-        """A PeekMessage cannot be completed Raises `TypeError`."""
-        raise TypeError("Peeked message cannot be completed.")
+    @property
+    def partition_id(self):
+        if self.message.annotations:
+            return self.message.annotations.get(_X_OPT_PARTITION_ID)
+        return None
 
-    def dead_letter(self, description=None):
-        """A PeekMessage cannot be dead-lettered. Raises `TypeError`."""
-        raise TypeError("Peeked message cannot be dead-lettered.")
+    @property
+    def enqueued_time(self):
+        if self.message.annotations:
+            timestamp = self.message.annotations.get(_X_OPT_ENQUEUED_TIME)
+            if timestamp:
+                in_seconds = timestamp/1000.0
+                return datetime.datetime.utcfromtimestamp(in_seconds)
+        return None
 
-    def abandon(self):
-        """A PeekMessage cannot be abandoned. Raises `TypeError`."""
-        raise TypeError("Peeked message cannot be abandoned.")
+    @property
+    def scheduled_enqueue_time(self):
+        if self.message.annotations:
+            timestamp = self.message.annotations.get(_X_OPT_SCHEDULED_ENQUEUE_TIME)
+            if timestamp:
+                in_seconds = timestamp/1000.0
+                return datetime.datetime.utcfromtimestamp(in_seconds)
+        return None
 
-    def defer(self):
-        """A PeekMessage cannot be deferred. Raises `TypeError`."""
-        raise TypeError("Peeked message cannot be deferred.")
+    @property
+    def sequence_number(self):
+        if self.message.annotations:
+            return self.message.annotations.get(_X_OPT_SEQUENCE_NUMBER)
+        return None
 
 
-class DeferredMessage(Message):
-    """A message that has been deferred.
+class ReceivedMessage(PeekMessage):
```

Comment:
I don't see `Message` as "outgoing message", I think it's meta class.
`ReceivedMessage` keeps all the attributes of meta class and provides settlement method.

Reply:
and if we are in the opposite direction, then PeekMessage would inherit "settlement" methods just like our Track1, which need to override to raise errors.

Reply:
Following up on this topic:
We want to keep the design for now (received message received outgoing message), this makes the received message re-sendable (it can be sent again)

Reply:
topic included in issue: https://github.com/Azure/azure-sdk-for-python/issues/10602 if further discussion is needed.